### ISSUE TITLE
Document the impact this proposal has on limit changes in the future.

### ISSUE THREAD

**Issue by monsterbitar** [ActorForth/Auction-Protocol#7]
Without the change to counting the op_return bytes in aggregate, a change in the maximum bytes allowed in an op_return output would remain the same as long as there is only a single op_return in a transaction.

If the restriction of a single op_return would be dropped from standardness but the size limit would remain, then a user could create a transaction with a large number of op_returns, effectively rendering the byte limitation ineffecient.

If multiple opreturns were to be allowed but in a limited number, then the byte limitation would become a multiple of that allowed number of opreturns, which would make entangle the byte limitation with the opreturn count limitation. In such a case, changing the byte limit would inadvertently get amplified.

Take for example, a change that would set the allowed number of opreturns to 64. Not an entirely unreasonable number, but after that change any increases or decreases to the opreturn byte limitation would be amplified to the point where selecting an intended targer number for total storage bytes, might no difficult. For example, the current size of 224b is not evenly divisible by 64 and would not be a candidate.

In short, assuming that Bitcoin Cash does allow multiple op_returns, the following outcomes can be observed:

Change | With aggregated counting | without aggregated counting
--- | --- | ---
smaller | any number is acceptable, but overhead from multiple outputs has a larger impact | selected limit is amplified, but overhead is fixed relative to the outputs.
larger | any number is acceptable, and overhead from multiple outputs matter less | selected limit is amplified, overhead remains a fixed cost relate to the outputs.
none | users can choose to pay additional overhead in order to get additional value | users can choose to use more outputs to store additional data.

**Comment by scherrey** [comment]
Regarding the change options:
1) smaller - I think ANY reduction in max sizes is a profound and breaking change that is difficult to fully comprehend. That's why we need to make size change decisions quite conservatively because once it's increased, there's really no practical way to bring it back in. 

2) larger or none - the CHIP takes an explicit position that it requires "aggregated counting". 

I've broken down what I think are the relevant considerations into the following paragraph and added it to the CHIP:

Without the aggregate size consideration, the future impact of changing the legal sizes of OP_RETURN becomes more profound and possibly difficult to predict but saves tx creators from having to track a total across all OP_RETURNs. The opinion of the author is that the days of being able to consider parts of transactions independent of others are long gone so this is not a meaningful shortcoming. Size reductions would be profound and dangerous changes and are unlikely to be practical to deploy. Size increases should be based on empirical demonstrations of value through experimental apps that push these limits while demonstrating real value to the BCH ecosystem. Regardless of the ultimate size, having that quantity come from a shared pool helps allow OP_RETURN based protocols to be expressed naturally in a manner more closely resembling what their ultimate core protocol implementations would look like rather than having to perform hacks to fit within a per-OP_RETURN limit that would necessarily be smaller. Again these are future considerations outside of the direct scope of this CHIP.
